src/js/manageList.js

```diff
@@ -85,29 +93,43 @@ export default class ManageList {
   }
 
   completed(id) {
-    const keys = Object.keys(localStorage);
-    keys.forEach((key) => {
-      this.taskList.push(JSON.parse(localStorage.getItem(key)));
-    });
-    this.taskList.forEach((task) => {
+    let retreived = JSON.parse(localStorage.getItem('tasks'));
+
+    retreived.forEach((task) => {
       if (id === task.index) {
         task.completed = true;
-        localStorage.setItem(`task${id}`, JSON.stringify(task));
       }
+      localStorage.setItem('tasks', JSON.stringify(retreived));
     });
+
+    this.listDescription.forEach((desc) => {
+      if (Number(desc.id) === id) {
+        desc.classList.add('striked');
+      }
+    })
   }
 
   cancelCompleted(id) {
-    const keys = Object.keys(localStorage);
-    keys.forEach((key) => {
-      this.taskList.push(JSON.parse(localStorage.getItem(key)));
-    });
-    this.taskList.forEach((task) => {
+    let retreived = JSON.parse(localStorage.getItem('tasks'));
+    retreived.forEach((task) => {
       if (id === task.index) {
         task.completed = false;
-        localStorage.setItem(`task${id}`, JSON.stringify(task));
       }
+      localStorage.setItem('tasks', JSON.stringify(retreived));
     });
+
+    this.listDescription.forEach((desc) => {
+      if (Number(desc.id) === id) {
+        desc.classList.remove('striked');
+      }
+    })
+  }
+
+  clearCompleted() {
+    let retreived = JSON.parse(localStorage.getItem('tasks'));
+
+    const filtered = retreived.filter(task => task.completed !== true);
+    localStorage.setItem('tasks', JSON.stringify(filtered));
   }
```

Comment:
Good job! but according to project requirements, we expect you to put this method in a new file, and import it as a module for better code isolation. Kindly make sure to export this to a new file.

![image](https://user-images.githubusercontent.com/34779928/206857826-199dc2af-4ae5-4c2f-bcd2-802fa07b8b22.png)

Reply:
Well noted. All the changes have been made.